api_app/views.py

```diff
@@ -82,7 +85,19 @@ def _multi_analysis_request(
     else:
         # fire celery task
         for index, job in enumerate(jobs):
-            runtime_configuration = runtime_configurations[index]
+            runtime_configuration = {}
+
+            for analyzer in job.analyzers_to_execute:
+                # Appending custom config to runtime configuration
+                config = CustomConfig.get_as_dict(
+                    user, CustomConfig.Type.ANALYZER, name=analyzer
+                ).get(analyzer, {})
+                if analyzer in runtime_configurations[index]:
+                    config = config | runtime_configurations[index][analyzer]
+                if config:
+                    runtime_configuration[analyzer] = config
```

Comment:
this is complex logic that requires proper testing. So please extract this part to a separate function and do the tests of what you want that this would behave.

Reply:
Most of the heavy lifting in done in `get_as_dict`. It's just 2 `if` conditions after that. Should I still attempt the refactor?

Reply:
yeah, my point is to extract this to allow it to be unit-tested. Otherwise it is just not possible.

I mean, we need to test most the new things that we do but when we do changes to already existing APIs we should be careful even more. This is why extracting this and creating a simple unit test which proves that it works as you intend it to and that the normal behavior is preserved, would really help us in maintaining the application.

Reply:
I could do that. But databases are widely varying in how they handle JSON Fields, and the explicit conversions will still remain (for type-checking purposes).